lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ApacheParquetValueDecoder.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader.decoders;
+
+import io.trino.parquet.reader.SimpleSliceInputStream;
+import org.apache.parquet.bytes.ByteBufferInputStream;
+import org.apache.parquet.column.values.ValuesReader;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+
+import static io.trino.parquet.ParquetReaderUtils.castToByte;
+import static io.trino.parquet.ParquetReaderUtils.toByteExact;
+import static io.trino.parquet.ParquetReaderUtils.toShortExact;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This is a set of proxy value decoders that use a delegated value reader from apache lib.
+ */
+public abstract class ApacheParquetValueDecoder<T>
```

Comment:
Reusing code via inheritance is generally not a good idea. It makes it harder to follow what the code is doing, introduces ambiguity with respect to whether implementations are supposed to call super() and when to do it, and a few other potential problems.

Remove it and make each of those decoders implement ValueDecoder directly. To reuse the implementation of the `init()` method, extract it into a static method somewhere that all those implementation can call.